keys/keys.go

```diff
@@ -0,0 +1,149 @@
+package keys
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/google/uuid"
+)
+
+type (
+	// Key represents a kafka key.
+	Key interface {
+		sarama.Encoder
+	}
+
+	// CreateTimer is a create timer or delete timer event, value should be a messaging.CreateTimer or nil.
+	CreateTimer struct {
+		domain    string
+		timerUUID uuid.UUID
+	}
+
+	// ExecuteTimer is an execute timer event, value should be a common.ExecuteTimer or nil.
+	ExecuteTimer struct {
+		domain    string
+		timerUUID uuid.UUID
+	}
+
+	// Dummy is a dummy event so a node knows when its caught up after a repartition. Value should be nil.
+	Dummy struct{}
```

Comment:
I am confused? Why does this exist? When do you write it?

Reply:
When a node joins and needs to wait for the "present" to start firing, so it knows that it's not firing timers that were deleted a long time ago.

Reply:
I should note that when we do that we also tack on a header specifying the generationID. Generation ID is like zxid of consumer membership circle I think. So even though all the keys just look like "dummy", we know which one we sent. There are other ways I could've done this. Since value is nil the keys could specify generationID and they'd still go away w/ compaction. I think i might do this actually so I don't have to bother with putting a header.

Reply:
Actually maybe not. I also might want to specify generationID in executetimer messages just for understanding what's going on better, i'll know if someone held on to a timer for some time after a rebalancing by detecting a decreasing generationID, and for that i'll want it in headers so that it compacts. Might make sense to put it in headers here too to be consistent.

Reply:
Also I tried to find other ways to solve the problem of finding the present and their is one other solution. It involves making a synchrounous request to one of the kafka brokers (you have to single one out) asking when the latest consumed message was.